src/types/github-types.ts

```diff
@@ -1,26 +1,88 @@
 import { RestEndpointMethodTypes } from "@octokit/rest";
 import { Context } from "./context";
 
-export type Issue = RestEndpointMethodTypes["issues"]["get"]["response"]["data"];
-export type IssueComments = RestEndpointMethodTypes["issues"]["listComments"]["response"]["data"][0];
-export type ReviewComments = RestEndpointMethodTypes["pulls"]["listReviewComments"]["response"]["data"][0];
+type BaseIssue = RestEndpointMethodTypes["issues"]["get"]["response"]["data"];
+export interface Issue extends BaseIssue {
+  prDetails?: PullRequestDetails;
+}
+
 export type User = RestEndpointMethodTypes["users"]["getByUsername"]["response"]["data"];
 
 export type FetchParams = {
   context: Context;
   issueNum?: number;
   owner?: string;
   repo?: string;
+  currentDepth?: number; // Current depth in the tree
+  maxDepth?: number; // Maximum depth to traverse
+  parentIssueKey?: string; // Parent issue key (Tree structure)
+};
+
+type Repository = {
+  owner: {
+    login: string;
+  };
+  name: string;
+};
+
+type IssueData = {
+  number: number;
+  url: string;
+  body: string;
+  repository: Repository;
+};
+
+type PullRequestNode = {
+  id: string;
+  body: string;
+  closingIssuesReferences: {
+    nodes: IssueData[];
+  };
+};
+
+type PullRequestReviewCommentNode = {
+  id: string;
+  body: string;
+  pullRequest: PullRequestNode;
 };
 
-export type LinkedIssues = {
+type IssueCommentNode = {
+  id: string;
+  body: string;
+  issue: IssueData;
+};
+
+export type GqlIssueSearchResult = {
```

Comment:
```suggestion
export type GraphQlIssueSearchResult = {
```

I would write out the full term to reduce cognitive overhead.